recipes/pdcurses/all/conanfile.py

```diff
@@ -0,0 +1,155 @@
+                args.append("DLL=Y")
+            args = " ".join(args)
+            if self.settings.compiler == "Visual Studio":
+                with tools.vcvars(self.settings):
+                    self.run("nmake -f Makefile.vc {}".format(args))
+            else:
+                self.run("{} libs {}".format(os.environ["CONAN_MAKE_PROGRAM"], args))
+
+    def _patch_sources(self):
+        if self.settings.compiler == "Visual Studio":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "wincon", "Makefile.vc"),
+                                  "$(CFLAGS)",
+                                  "$(CFLAGS) -{}".format(self.settings.compiler.runtime))
+            tools.replace_in_file(os.path.join(self._source_subfolder, "wincon", "Makefile.vc"),
+                                  "$(LDFLAGS)",
+                                  "$(LDFLAGS) -{}".format(self.settings.compiler.runtime))
+        tools.replace_in_file(os.path.join(self._source_subfolder, "x11", "Makefile.in"),
+                              "$(INSTALL) -c -m 644 $(osdir)/libXCurses.a $(libdir)/libXCurses.a",
+                              "-$(INSTALL) -c -m 644 $(osdir)/libXCurses.a $(libdir)/libXCurses.a")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "x11", "Makefile.in"),
+                              "\nall:\t",
+                              "\nall:\t{}\t#".format("@SHL_TARGETS@" if self.options.shared else "$(LIBCURSES)"))
+
+    def build(self):
+        self._patch_sources()
+        if self.settings.os == "Windows":
+            self._build_windows()
+        else:
+            with tools.chdir(os.path.join(self._source_subfolder, "x11")):
+                autotools = self._configure_autotools()
+                autotools.make()
+
+    @property
+    def _subsystem_folder(self):
+        return {
+            "Windows": "wincon",
+        }.get(str(self.settings.os), "x11")
+
+    @property
+    def _license_text(self):
+        readme = tools.load(os.path.join(self._source_subfolder, self._subsystem_folder, "README.md"))
+        match = re.search("Distribution Status\n[\\-]+(?:[\r\n])+((?:[0-9a-z .,;*]+[\r\n])+)", readme,
+                          re.IGNORECASE | re.MULTILINE)
+        if not match:
+            raise ConanException("Cannot extract distribution status")
+        return match.group(1).strip() + "\n"
+
+    def package(self):
+        tools.save(os.path.join(self.package_folder, "licenses", "LICENSE"), self._license_text)
+
+        if self.settings.os == "Windows":
+            self.copy(pattern="curses.h", src=self._source_subfolder, dst="include")
+            self.copy(pattern="*.dll", dst="bin", keep_path=False)
+            self.copy(pattern="*.lib", dst="lib", keep_path=False)
+            self.copy(pattern="*.a", dst="lib", keep_path=False)
+
+            if self.settings.compiler != "Visual Studio":
+                os.rename(os.path.join(self.package_folder, "lib", "pdcurses.a"),
```

Comment:
why does os.rename should work for self.options.shared?

Reply:
Can you rephrase your question? I don't understand it :smile:

Reply:
ping @uilianries

Reply:
This renaming only happens on Windows.
On Windows, even dll's have a static (imp) library.

Reply:
5 days after my comment, I can't realize what I tried to say ... Anyway, it looks good.